### ISSUE TITLE
Roslyn is more strict than Visual Studio when compiling project

### ISSUE THREAD

**Issue by aka-STInG** [dotnet/roslyn#16263]
**Version Used**: 
1.3.2
**Steps to Reproduce**:

1. Project A.csproj contains T4 template which produces code file FancyClass1.cs I had code file added both in project and in shared project, in VS everything compiles ok without warnings, with roslyn I get a bunch of errors about duplicate declarations
2. Shared project B.shproj has FancyClass1.cs added as link
3. Project A.csproj loaded with MSBuildWorspace and compiled with Project.GetCompilationAsync()
**Expected Behavior**:
Same behavior as Visual Studio: compilation without errors (or vise versa).
In other words: Consistent behavior between Visual Studio and roslyn compilation.
**Actual Behavior**:
In compilation there are diagnostics with error severity about type duplication.

**Comment by gafter** [comment]
Can you please package up and provide a complete solution that exhibits the issue?

**Comment by aka-STInG** [comment]
[RoslynBug16263Repro.docx](https://github.com/dotnet/roslyn/files/688480/RoslynBug16263Repro.docx)
It is a zip, for some reason github does not allow it, although states otherwise

**Comment by aka-STInG** [comment]
[RoslynBug16263Repro.zip](https://github.com/dotnet/roslyn/files/689422/RoslynBug16263Repro.zip)

**Comment by mattwar** [comment]
I suspect this problem is already fixed. 

I've tried this using most recent roslyn package from roslyn myget feed and using msbuild 15 (VS2017).  The type duplicate error does not occur.  I only see one compilation error and it is about there not being a Main method entry point.
